### PR TITLE
[py-gpustat] Added new version

### PR DESCRIPTION
**Pull request by mdorier**
Added new version for the py-gpustat package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-gpustat/package.py [hunk 1]
```diff
@@ -10,6 +10,7 @@ class PyGpustat(PythonPackage):
     homepage = "https://github.com/wookayin/gpustat"
     pypi = "gpustat/gpustat-0.6.0.tar.gz"
 
+    version('1.0.0b1', sha256='a25c460c5751180265814f457249ba5100baf7a055b23ad762a4e3ab3f6496dd')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gpustat/package.py:13]: This is a pre-release, we should mark 0.6.0 as `preferred=True`

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Dependencies also need to be updated, see `setup.py`

**Comment by mdorier** [comment]
I updated the dependencies, now I think I need to make a PR to add more versions to `py-blessed`...

File: var/spack/repos/builtin/packages/py-gpustat/package.py [hunk 2]
```diff
@@ -10,12 +10,15 @@ class PyGpustat(PythonPackage):
     homepage = "https://github.com/wookayin/gpustat"
     pypi = "gpustat/gpustat-0.6.0.tar.gz"
 
-    version('0.6.0', sha256='f69135080b2668b662822633312c2180002c10111597af9631bb02e042755b6c')
+    version('1.0.0b1', sha256='a25c460c5751180265814f457249ba5100baf7a055b23ad762a4e3ab3f6496dd')
+    version('0.6.0', sha256='f69135080b2668b662822633312c2180002c10111597af9631bb02e042755b6c',
+            preferred=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gpustat/package.py:16]: Latest version requires Python 3.4+

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-gpustat/package.py (hunk 2 above)
Comment (mdorier) [reply var/spack/repos/builtin/packages/py-gpustat/package.py:16]: I updated it. With the new versions of py-blessed now upstream, I believe we have everything necessary for this PR to be merged now.

File: var/spack/repos/builtin/packages/py-gpustat/package.py [hunk 3]
```diff
@@ -10,12 +10,16 @@ class PyGpustat(PythonPackage):
...
 
+    depends_on('python@3.4:', when='@1.0.0:', type=('build', 'run'))
     depends_on('py-setuptools', type='build')
     depends_on('py-pytest-runner', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gpustat/package.py]: Can you add:
```python
depends_on('py-pytest-runner@5:', when='@1: ^python@3.5:', type='build')
depends_on('py-pytest-runner@:4', when='@1: ^python@:3.4', type='build')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**